compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2115,7 +2115,7 @@ impl<'tcx> TyCtxt<'tcx> {
             }
         }
 
-        Lrc::new(set)
+        set.into_iter()
```

Comment:
What I actually meant was to change this to something using `std::iter::from_fn`, like:

```rust
let mut set = FxHashSet::default();
let mut stack = vec![trait_def_id];
set.insert(trait_def_id);
std::iter::from_fn(move || -> Option<DefId> {
    let trait_did = stack.pop()?;
    let generic_predicates = self.super_predicates_of(trait_did);
    for each predicate {
        if set.insert(data.def_id()) { stack.push(data.def_id()); }
    }

    Some(trait_did)
})
```

this way, we won't build up the whole vector, but instead we'll stop as soon as we find a match

Reply:
Done!